include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -95,11 +95,6 @@ class format_sam_base
                                  header_type      & header,
                                  ref_seqs_type    & /*tag*/);
 
-    static void update_alignment_lengths(int32_t & ref_length,
```

Comment:
This static void does not appear in the `include/seqan3/io/sam_file/detail/cigar.hpp`. Is it just unnecessary ? :)

Reply:
As the function is not in a `class` anymore, I think the `static` is not necessary.

Reply:
A `static` member of a class would mean that the member (in this case a member function) is independent of the actual object. Maybe think of it as "there is one (global) instance of this member" and the member exists without any instance of the class existing.

It also means that you do not need an instance of the `format_sam_base` (or any derived) class to invoke this function. You can use `format_sam_base::update_alignment_lengths` instead of `some_instance.update_alignment_lengths`.